tests/sles4sap/netweaver_patch.pm

```diff
@@ -0,0 +1,93 @@
+# SUSE's openQA tests
+#
+# Copyright 2017-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Perform an unattended installation of SAP NetWeaver
+# Requires: ENV variable NW pointing to installation media
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use base "sles4sap";
+use testapi;
+use lockapi;
+use hacluster;
+use strict;
+use warnings;
+
+sub run {
+    my ($self) = @_;
+    my ($proto, $path) = $self->fix_path(get_required_var('NW'));
+    my $instance_type = get_required_var('INSTANCE_TYPE');
+    my $instance_id = get_required_var('INSTANCE_ID');
+    my $sid = get_required_var('INSTANCE_SID');
+    my $arch = get_required_var('ARCH');
+    my $hostname = get_var('INSTANCE_ALIAS', '$(hostname)');
+    my $params_file = "/sapinst/$instance_type.params";
+    my $timeout = 900 * get_var('TIMEOUT_SCALE', 1);    # Time out for NetWeaver's sources related commands
+    my $product_id = undef;
+    my $pscmd = $self->set_ps_cmd(get_required_var('INSTANCE_TYPE'));
+    my $sap_dir = "/usr/sap/$sid";
+    my $cluster_name = undef;
+    $cluster_name = get_cluster_name if get_var('HA_CLUSTER');
+
+    # Set Product ID depending on the type of Instance
+    if ($instance_type eq 'ASCS') {
+        $product_id = 'NW_ABAP_ASCS';
+    }
+    elsif ($instance_type eq 'ERS') {
+        $product_id = 'NW_ERS';
+    }
```

Comment:
An `else` may be added to fail on wrong instance type.

Reply:
Very good point, but I've added the else in the netweaver_install.pm module (that is always running before this one).